getgauge/refactor.py

```diff
@@ -18,7 +19,7 @@ def refactor_step(request, response):
     )
     content = impl_file.get_code()
     if request.saveChanges:
-        with open(info.file_name, 'w') as f:
+        with open(info.file_name, 'w', encoding="utf-8") as f:
```

Comment:
Potentially dumb question, but could this potentially be a problem where folks have existing impl files that are on Windows with `CP-1252` if using some system encoding was the default earlier?

Or has this default changed long enough ago within Python on Windows that we've been opening them assuming utf-8 already?

Reply:
I think the question is not dumb at all. If gauge users are not "carefully" enough which encoding is used in their projects, they may encounter issues with this change. I introduced this "encoding" attribute also in the other PRs.

Should I revert this explicit encoding so that gauge-python is backwards compatible?

In general, this resolves the following PyLint warning: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/unspecified-encoding.html

Reply:
Perhaps I wasn't paying close enough attention, but I had thought earlier the encodings clarified were for things within the build rather than user files at runtime - but might have missed the `read_file_contents` one when getting content to implement a stub that potentially could break in some cases.

However after a bit of digging, I do note https://github.com/getgauge/gauge-python/commit/b13ca190e91efebed8dd610d747a41760bc572db which was changed summarily in the most critical place during parsing for step execution, so your change is probably fine, and at least makes things consistent....

(if we were being super paranoid and wanted to get rid of the warning, `encoding="locale"` is probably the only fully backward compatible option - but Python `3.10` only - and doesn't get to the root of the problem anyway)